command/harness/delegate/delegate.go

```diff
@@ -0,0 +1,250 @@
+package delegate
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/drone-runners/drone-runner-aws/command/config"
+	"github.com/drone-runners/drone-runner-aws/command/harness"
+	"github.com/drone-runners/drone-runner-aws/engine/resource"
+	"github.com/drone-runners/drone-runner-aws/internal/drivers"
+	"github.com/drone-runners/drone-runner-aws/internal/httprender"
+	errors "github.com/drone-runners/drone-runner-aws/internal/types"
+	"github.com/drone-runners/drone-runner-aws/store"
+	"github.com/drone-runners/drone-runner-aws/store/database"
+	loghistory "github.com/drone/runner-go/logger/history"
+	"github.com/drone/runner-go/server"
+	"github.com/drone/signal"
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/joho/godotenv"
+	"github.com/sirupsen/logrus"
+	"github.com/wings-software/dlite/logger"
+	"golang.org/x/sync/errgroup"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+type delegateCommand struct {
+	envFile         string
+	env             config.EnvConfig
+	poolFile        string
+	poolManager     *drivers.Manager
+	stageOwnerStore store.StageOwnerStore
+}
+
+func (c *delegateCommand) delegateListener() http.Handler {
+	mux := chi.NewMux()
+
+	mux.Use(func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			wrap := middleware.NewWrapResponseWriter(w, r.ProtoMajor)
+
+			reqStart := time.Now().UTC()
+			next.ServeHTTP(wrap, r)
+
+			status := wrap.Status()
+			dur := time.Since(reqStart).Milliseconds()
+
+			logr := logrus.WithContext(r.Context()).
+				WithField("t", reqStart.Format(time.RFC3339)).
+				WithField("status", status).
+				WithField("dur[ms]", dur)
+			logLine := "HTTP: " + r.Method + " " + r.URL.RequestURI()
+			if status >= http.StatusInternalServerError {
+				logr.Errorln(logLine)
+			} else {
+				logr.Infoln(logLine)
+			}
+		})
+	})
+
+	mux.Post("/pool_owner", c.handlePoolOwner)
+	mux.Post("/setup", c.handleSetup)
+	mux.Post("/destroy", c.handleDestroy)
+	mux.Post("/step", c.handleStep)
+
+	return mux
+}
+
+func RegisterDelegate(app *kingpin.Application) {
+	c := new(delegateCommand)
+
+	c.poolManager = &drivers.Manager{}
+
+	cmd := app.Command("delegate", "starts the delegate").
+		Action(c.run)
+	cmd.Flag("envfile", "load the environment variable file").
+		StringVar(&c.envFile)
+	cmd.Flag("pool", "file to seed the pool").
+		StringVar(&c.poolFile)
+}
+
+func (c *delegateCommand) run(*kingpin.ParseContext) error {
```

Comment:
remove instances in case of SIGTERM call and reuse not set. can be taken up in next pr.